.circleci/config.yml

```diff
@@ -0,0 +1,183 @@
+version: 2
+workflows:
+  version: 2
+  tests:
+    jobs: &workflow_jobs
+      - node6:
+          filters: &all_commits
+            tags:
+              only: /.*/
+      - node8:
+          filters: *all_commits
+      - node10:
+          filters: *all_commits
+      - lint:
+          requires:
+            - node6
+            - node8
+            - node10
+          filters: *all_commits
+      - docs:
+          requires:
+            - node6
+            - node8
+            - node10
+          filters: *all_commits
+      - system_tests:
+          requires:
+            - lint
+            - docs
+          filters: &master_and_releases
+            branches:
+              only: master
+            tags: &releases
+              only: '/^v[\d.]+$/'
+      - sample_tests:
+          requires:
+            - lint
+            - docs
+          filters: *master_and_releases
+      - publish_npm:
+          requires:
+            - system_tests
+            - sample_tests
+          filters:
+            branches:
+              ignore: /.*/
+            tags: *releases
+  nightly:
+    triggers:
+      - schedule:
+          cron: 0 7 * * *
+          filters:
+            branches:
+              only: master
+    jobs: *workflow_jobs
+jobs:
+  node6:
+    docker:
+      - image: 'node:6'
+        user: node
+    steps: &unit_tests_steps
+      - checkout
+      - run: &remove_package_lock
+          name: Remove package-lock.json if needed.
+          command: |
+            WORKFLOW_NAME=`python .circleci/get_workflow_name.py`
+            echo "Workflow name: $WORKFLOW_NAME"
+            if [ "$WORKFLOW_NAME" = "nightly" ]; then
+              echo "Nightly build detected, removing package-lock.json."
+              rm -f package-lock.json samples/package-lock.json
+            else
+              echo "Not a nightly build, skipping this step."
+            fi
+      - run: &npm_install_and_link
+          name: Install and link the module
+          command: |-
+            mkdir -p /home/node/.npm-global
```

Comment:
Since we don't do `npm link` here, there is no need to `mkdir`, and no need to set `NPM_CONFIG_PREFIX` below in line 80.

Reply:
This file is getting auto-generated by synthtool now.  I really want to avoid custom stuff in these circleci configs, especially as we get ready to migrate to kokoro.

Reply:
Sounds reasonable - and since it does no harm we can as well leave it here (I hope that since we're going to migrate to kokoro, we are not planning any massive CircleCI config changes).